packages/shared/lib/services/config.service.ts

```diff
@@ -21,10 +21,25 @@ class ConfigService {
     public DEMO_GITHUB_CONFIG_KEY = 'demo-github-integration';
 
     private getTemplatesFromFile() {
-        const templatesPath = path.join(dirname(), '../../../providers.yaml');
+        const templatesPath = () => {
+            // find the providers.yaml file
+            // recursively searching in parent directories
+            const findProvidersYaml = (dir: string): string => {
+                const providersYamlPath = path.join(dir, 'providers.yaml');
+                if (fs.existsSync(providersYamlPath)) {
+                    return providersYamlPath;
+                }
+                const parentDir = path.dirname(dir);
+                if (parentDir === dir) {
+                    throw new NangoError('providers_yaml_not_found');
+                }
+                return findProvidersYaml(parentDir);
+            };
+            return findProvidersYaml(dirname());
+        };
```

Comment:
When called from tests, dirname is actually different and tests cannot find the provider. Since it walks backward from the current dir it could find unrelated providers.yaml but only if our is not present

Reply:
Thanks for fixing this!